lib/typing/columns/default.go

```diff
@@ -12,28 +12,23 @@ import (
 	"github.com/artie-labs/transfer/lib/typing/decimal"
 )
 
-type DefaultValueArgs struct {
-	Escape   bool
-	DestKind constants.DestinationKind
-}
-
 func (c *Column) RawDefaultValue() any {
 	return c.defaultValue
 }
 
-func (c *Column) DefaultValue(args *DefaultValueArgs, additionalDateFmts []string) (any, error) {
-	if args == nil || !args.Escape || c.defaultValue == nil {
+func (c *Column) DefaultValue(dialect sql.Dialect, additionalDateFmts []string) (any, error) {
+	if c.defaultValue == nil {
 		return c.defaultValue, nil
 	}
 
 	switch c.KindDetails.Kind {
 	case typing.Struct.Kind, typing.Array.Kind:
-		switch args.DestKind {
-		case constants.BigQuery:
+		switch dialect.(type) {
+		case sql.BigQueryDialect:
 			return "JSON" + stringutil.Wrap(c.defaultValue, false), nil
-		case constants.Redshift:
+		case sql.RedshiftDialect:
 			return fmt.Sprintf("JSON_PARSE(%s)", stringutil.Wrap(c.defaultValue, false)), nil
-		case constants.Snowflake:
+		case sql.SnowflakeDialect:
```

Comment:
@Tang8330 it doesn't seem like we're doing anything for MS SQL here, do we need a case for it?

Reply:
Yeah we're not. We don't support backfill default values in MSSQL yet.

Do you want to leave a comment?